### DISCUSSION TITLE
How to pass more than one UNIQUE columns into DoUpdate::target?

### DISCUSSION THREAD

**Discussion by zmeid** [simolus3/drift#1304] [Q&A] [answered]
DoUpdate::target accepts array of columns. But, if we pass more than one column, it treats them as group of unique columns. 

Let's say i have a table with -> A, B, C, D columns. Constraints: Unique(A, B), Unique(C, D).

What if i want it to do update when there will be conflict on Unique(A, B) or Unique(C, D)? When i pass A,B into target[], it will do update if there is conflict on Unique(A, B) which makes sense, but, how to also add Unique(C, D) into target?

**Comment by simolus3** [comment, marked as answer]
Having multiple upsert clauses is a fairly new sqlite3 feature that wasn't available when I wrote the original upsert api in Dart.

In a1c9a5933d6324fb161222f3bb0bcdfc041e613a, I introduced the `UpsertMultiple` class which you can use to apply multiple `DoUpdate` clauses.

**Reply by zmeid** [reply]
That's great! Any idea when that would be released?

**Reply by simolus3** [reply]
I'll try to release it this weekend.
